### PR TITLE
Update tests

### PR DESCRIPTION
**Pull request by SwampDragons**
- Add a makefile
- refactor code to move away from the RenderConfig() usage, and just use fully-written example templates in the test-fixtures folder. 
- Upgrade tests to use the newer testing framework from the plugin sdk that shells out to `packer build` instead of creating and running a core. This allows us to upgrade to the version of the SDK that strips out the circular dependency with Packer, once we've released it. 
- use hcl2_upgrade to create HCL versions of the acceptance tests we already run; and add those files to run in addition to the JSON tests.

### CODE REVIEW

**Comment by SwampDragons** [comment]
The linter is failing but I didn't actually change the file being linted and it feels outside the scope of this PR.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,27 @@
+
+NAME=scaffolding
```
Comment (nywilken) [inline Makefile:2]:
```suggestion
NAME=docker
```

File: builder/docker/communicator_test.go [hunk 2]
```diff
@@ -2,104 +2,72 @@ package docker
...
+				}
+				if sha256.Sum256(inputFile) != sha256.Sum256(outputFile) {
+					return fmt.Errorf("Input and output files do not match\n"+
+						"Input:\n%s\nOutput:\n%s\n", inputFile, outputFile)
+				}
+				return nil
+			},
+			Teardown: func() error {
+				// Cleanup. Honestly I don't know why you would want to get rid
+				// of my strawberry cake. It's so tasty! Do you not like cake? Are you a
```
Comment (nywilken) [inline builder/docker/communicator_test.go:64]: 🤣

**Review by nywilken — APPROVED** [review]
This looks good. Nicely done. 

I think the lint errors are showing up there because this is the first time it has been executed against in the repo. There is also no config file so all errors are being counted now. This is a small set that we should fix.

File: builder/docker/communicator_test.go (hunk 2 above)
Comment (sylviamoss) [reply builder/docker/communicator_test.go:64]: What happened here? 👀  haha

**Comment by sylviamoss** [comment]
Nice to see the tests with the new framework 🎉

File: builder/docker/communicator_test.go [hunk 3]
```diff
@@ -2,104 +2,72 @@ package docker
...
+				// cake-hater? Or are you keeping all the cake all for yourself? So selfish!
+				os.Remove("my-strawberry-cake")
```
Comment (azr) [inline builder/docker/communicator_test.go:66]: 😆

**Pull request merged**